fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/RecordCursorTest.java

```diff
@@ -1180,4 +1180,15 @@ public void hasNextErrorStack() throws Exception {
         }
     }
 
+    @Test
+    public void reduceTest() throws Exception {
+        RecordCursor<Integer> cursor = RecordCursor.fromList(Arrays.asList(1, 2, 3, 4, 5, 6, 7));
+        CompletableFuture<Integer> sum = cursor.reduce(0, Integer::sum);
+        assertEquals(28, sum.get());
+
+        cursor = RecordCursor.fromList(Arrays.asList(1, 2, 3, 4, 5, 6, 7));
+        sum = cursor.reduce(0, Integer::sum, x -> x > 10);
```

Comment:
Need a variant of this test where the stop condition is never met and it goes to the end. I am pretty sure that would uncover the problem mentioned above.

Reply:
Done, thanks!